this patch is derived from:

 From fdc9866e48797f22762da457688b6ca93b6a337e Mon Sep 17 00:00:00 2001
 From: Takuro Ashie <ashie@homa.ne.jp>
 Date: Wed, 8 May 2019 14:42:24 +0900
 Subject: [PATCH] Disable eglQueryString(nullptr, EGL_EXTENSIONS)
 
 Because it causes crash on certain environments such as PowerVR drivers,
 and such extensions aren't used on Linux in fact.
 
 Upstream-Status: Inappropriate [OE-specific: https://bugzilla.mozilla.org/show_bug.cgi?id=1571603]
 
 Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
 Signed-off-by: Takuro Ashie <ashie@clear-code.com>
 
 ---
  gfx/gl/GLContextProviderEGL.cpp | 6 ++++++
  gfx/gl/GLLibraryEGL.cpp         | 2 ++
  2 files changed, 8 insertions(+)

Index: firefox-84.0.2/gfx/gl/GLContextProviderEGL.cpp
===================================================================
--- firefox-84.0.2.orig/gfx/gl/GLContextProviderEGL.cpp
+++ firefox-84.0.2/gfx/gl/GLContextProviderEGL.cpp
@@ -587,11 +587,13 @@ void GLContextEGL::GetWSIInfo(nsCString*
   out->Append((const char*)mEgl->mLib->fQueryString(mEgl->mDisplay,
                                                     LOCAL_EGL_EXTENSIONS));
 
+#if 0
 #ifndef ANDROID  // This query will crash some old android.
   out->AppendLiteral("\nEGL_EXTENSIONS(nullptr): ");
   out->Append(
       (const char*)mEgl->mLib->fQueryString(nullptr, LOCAL_EGL_EXTENSIONS));
 #endif
+#endif
 }
 
 // hold a reference to the given surface
Index: firefox-84.0.2/gfx/gl/GLLibraryEGL.cpp
===================================================================
--- firefox-84.0.2.orig/gfx/gl/GLLibraryEGL.cpp
+++ firefox-84.0.2/gfx/gl/GLLibraryEGL.cpp
@@ -792,6 +792,7 @@ void GLLibraryEGL::InitLibExtensions() {
 
   const char* rawExtString = nullptr;
 
+#if 0
 #ifndef ANDROID
   // Bug 1209612: Crashes on a number of android drivers.
   // Ideally we would only blocklist this there, but for now we don't need the
@@ -799,6 +800,7 @@ void GLLibraryEGL::InitLibExtensions() {
   // rather not crash.
   rawExtString = (const char*)fQueryString(nullptr, LOCAL_EGL_EXTENSIONS);
 #endif
+#endif
 
   if (!rawExtString) {
     if (shouldDumpExts) {
